SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -485,11 +490,33 @@ bool FLocalDeploymentManager::GetServiceStatus()
 		bLocalDeploymentRunning = false;
 		return false;
 	}
-	else
+	else if (ServiceStatusResult.Contains(TEXT("Local API service is running")))
 	{
-		UE_LOG(LogSpatialDeploymentManager, Verbose, TEXT("Spatial service running."));
-		bSpatialServiceRunning = true;
-		return true;
+		// Get the project file path and ensure it matches the one for the currently running project.
+		FString SpatialServicePath;
+		if (ServiceStatusResult.Split(TEXT("project file path: "), nullptr, &SpatialServicePath))
+		{
+			// Remove the trailing '" cli-version' that comes with non-interactive 'spatial' calls.
+			SpatialServicePath.Split(TEXT("\" cli-version"), &SpatialServicePath, nullptr);
+
+			FString CurrentProjectSpatialPath = FPaths::Combine(FSpatialGDKServicesModule::GetSpatialOSDirectory(), TEXT("spatialos.json"));
+			FPaths::NormalizeDirectoryName(SpatialServicePath);
+			FPaths::RemoveDuplicateSlashes(SpatialServicePath);
+			if (CurrentProjectSpatialPath == SpatialServicePath)
```

Comment:
Probably it's fine, but should we care about case? Maybe use FString::Equal with ESearchCase::IgnoreCase?

Reply:
Cool idea, done